### PR TITLE
add mining points

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
smelting ores accumulates mining points, using tg values (probably needs to be rebalanced)
you can then claim these points to your id card (and definitely not steal other points from other miners' ore...)

## Why / Balance
a sense of pride and accomplishment

future point shops for gamer gear like modkits better armour etc to incentivise salv to mine besides a funny number they cant even see ingame

may be worth considering how upgraded oreproc apply to mining points since miners might refuse to smelt ore until its upgraded. tg removed upgrades (https://github.com/tgstation/tgstation/pull/53073) for basically this reason

## Technical details
- `MiningPointsComponent` stores points on id card / oreproc
- `MiningPointsLatheComponent` makes recipes with MiningPoints > 0 add points, allows you to claim points from an oreproc

## Media
regular vs industrial oreprocs smelting a stack of uranium ore

## Requirements
- [X] I have tested all added content and changes.
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes
no

**Changelog**
:trollface: worthless reddit money

### CODE REVIEW

File: Content.Client/Lathe/UI/LatheMenu.xaml [hunk 1]
```diff
@@ -132,9 +132,14 @@
...
+            <BoxContainer Orientation="Horizontal" Name="MiningPointsContainer" Visible="False">
+                <Label Name="MiningPointsLabel" HorizontalExpand="True"/>
+                <Button Name="MiningPointsClaimButton" Text="{Loc 'lathe-menu-mining-points-claim-button'}"/>
+            </BoxContainer>
+            <!-- End DeltaV Additions: Mining points -->
         </BoxContainer>
         </BoxContainer>
 
     </BoxContainer>
-
```
Comment (MilonPL) [inline Content.Client/Lathe/UI/LatheMenu.xaml]: untouch

File: Content.Client/Lathe/UI/LatheMenu.xaml.cs [hunk 2]
```diff
@@ -70,9 +80,29 @@ public void SetEntity(EntityUid uid)
...
+
         MaterialsList.SetOwner(Entity);
     }
 
+    /// <summary>
+    /// DeltaV: Updates the UI elements for mining points.
+    /// </summary>
+    private void UpdateMiningPoints(uint points)
+    {
+        MiningPointsClaimButton.Disabled = points == 0 || _player.LocalSession?.AttachedEntity is not {} player || _miningPoints.TryFindIdCard(player) == null;
```
Comment (MilonPL) [inline Content.Client/Lathe/UI/LatheMenu.xaml.cs]: this line is extremely long, move the last OR operator to a new line at least

File: Content.Shared/DeltaV/Salvage/Systems/MiningPointsSystem.cs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    #endregion
+    #region Public API
+
+    /// <summary>
+    /// Tries to find the user's id card and gets its <see cref="MiningPointsComponent"/>.
+    /// </summary>
+    /// <remarks>
+    /// Component is nullable for easy usage with the API due to Entity<T> not being usable for Entity<T?> arguments.
```
Comment (MilonPL) [inline Content.Shared/DeltaV/Salvage/Systems/MiningPointsSystem.cs]: use `&lt;` for left angle brace and `&gt;` for right angle brace or else you get this peak documentation

image

File: Content.Shared/DeltaV/Salvage/Systems/MiningPointsSystem.cs [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            return false;
+
+        ent.Comp.Points -= amount;
+        Dirty(ent);
+        return true;
+    }
+
+    /// <summary>
+    /// Add points to a holder, inflating the economy.
+    /// <summary>
```
Comment (MilonPL) [inline Content.Shared/DeltaV/Salvage/Systems/MiningPointsSystem.cs]: pro

File: Resources/Prototypes/Entities/Structures/Machines/lathe.yml [hunk 5]
```diff
@@ -1383,6 +1387,7 @@
     materialUseMultiplier: 0.75
     timeMultiplier: 0.5
     staticRecipes:
+      - BluespaceCrystal #Nyano - Summary: Bluespace Crystals can be created here.
```
Comment (MilonPL) [inline Resources/Prototypes/Entities/Structures/Machines/lathe.yml]: nyano fake just mark it as deltav

**Review by MilonPL — CHANGES_REQUESTED** [review]

File: Content.Shared/DeltaV/Salvage/Systems/MiningPointsSystem.cs (hunk 3 above)
Comment (deltanedas) [reply Content.Shared/DeltaV/Salvage/Systems/MiningPointsSystem.cs]: i love xml

File: Content.Shared/DeltaV/Salvage/Systems/MiningPointsSystem.cs (hunk 4 above)
Comment (deltanedas) [reply Content.Shared/DeltaV/Salvage/Systems/MiningPointsSystem.cs]: sleepy joe

File: Content.Shared/DeltaV/Salvage/Systems/MiningPointsSystem.cs [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+            return false;
+
+        ent.Comp.Points -= amount;
+        Dirty(ent);
+        return true;
+    }
+
+    /// <summary>
+    /// Add points to a holder.
+    /// <summary>
```
Comment (MilonPL) [inline Content.Shared/DeltaV/Salvage/Systems/MiningPointsSystem.cs]: close the xml smh

**Review by MilonPL — CHANGES_REQUESTED** [review]

**Review by MilonPL — APPROVED** [review]

**Comment by MilonPL** [comment]
can be balanced once it does anything more than being a number

**Pull request merged**